File: 02_FizzBuzz/FizzBuzz.java
class FizzBuzz {

    public static void main(String[] args) {

        FizzBuzz fizzBuzz = new FizzBuzz();
        fizzBuzz.printFizzBuzz();
    }

    public void printFizzBuzz() {
        int n = 100;

        for (int i = 1; i <= n; i++) {

            if (i % 3 == 0 && i % 5 == 0) {
                System.out.print("FizzBuzz");
            } else if (i % 3 == 0) {
                System.out.print("Fizz");
            } else if (i % 5 == 0) {
                System.out.print("Buzz");
            } else {
                System.out.print(i);
            }

            System.out.print(", " );
        }

    }

}

File: fishbus/FizzBuzz.cs
namespace Fishbus;

public static class FizzBuzz
{
    public static bool TryConvertNumberToWord(int number, out string outcome)
    {
        bool canConvertToFizz = Fizz.CanConvert(number);
        bool canConvertToBuzz = Buzz.CanConvert(number);

        if (canConvertToBuzz && canConvertToFizz)
        {
            outcome = Fizz.Value + Buzz.Value;
            return true;
        }
        else if (canConvertToFizz)
        {
            outcome = Fizz.Value;
            return true;
        }
        else if (canConvertToBuzz)
        {
            outcome = Buzz.Value;
            return true;
        }

        outcome = "";
        return false;
    }
}